File: contain_all_rots.js
// https://www.codewars.com/kata/5700c9acc1555755be00027e
function containAllRots(strng, arr) {
  if(strng === '') {
    return true;
  }
  const rotations = [];
  let len = strng.length - 1;
  for(let i = 0; i < len + 1; i++) {
    let j = 0;
    let tmp = strng[len];
    while(j < len) {
      tmp += strng[j];
      j++;
    }
    rotations.push(tmp);
    strng = tmp;
  }
  const findAll = rotations.filter(rotation => !arr.includes(rotation));
  return !findAll.length;
}

File: contain_all_rots.py
# https://www.codewars.com/kata/5700c9acc1555755be00027e
def contain_all_rots(strng, arr):
    if strng == '':
        return True
    rotations = []
    for i in strng:
        j = 0
        tmp = strng[len(strng) - 1]
        while(j < len(strng) - 1):
            tmp += strng[j]
            j += 1
        rotations.append(tmp)
        strng = tmp
    findAll = all(i in arr for i in rotations)
    return findAll
